src/script/components/MessagesList/Message/FailedToAddUsersMessage.tsx

```diff
@@ -178,24 +207,8 @@ const FailedToAddUsersMessage: React.FC<FailedToAddUsersMessageProps> = ({
         </p>
       </div>
       <div className="message-body" css={{flexDirection: 'column'}}>
-        {isOpen && (
-          <>
-            {message.reason === AddUsersFailureReasons.UNREACHABLE_BACKENDS && (
-              <>
-                {Object.entries(groupedUsers).map(([domain, domainUsers]) => (
-                  <MessageDetails key={domain} domain={domain} message={message} users={domainUsers}>
-                    {learnMore}
-                  </MessageDetails>
-                ))}
-              </>
-            )}
-            {message.reason === AddUsersFailureReasons.NON_FEDERATING_BACKENDS && (
-              <MessageDetails message={message} users={users}>
-                {learnMore}
-              </MessageDetails>
-            )}
-          </>
-        )}
+        {isOpen && <DetailsMessageComponent users={users}>{learnMore}</DetailsMessageComponent>}
```

Comment:
I have extracted those into 2 separate components (and then the `grouping` will only happen lazily if we have an unreachable backend error)